pulp_ansible/tests/functional/api/collection/v3/test_collection.py

```diff
@@ -0,0 +1,236 @@
+# coding=utf-8
+"""Tests related to upload of collections."""
+import hashlib
+import logging
+import re
+import unittest
+from urllib.parse import urljoin
+
+import pytest
+import requests
+
+from pulp_smash import api, config
+from pulp_smash.pulp3.utils import delete_orphans
+from pulp_smash.utils import http_get
+from requests.exceptions import HTTPError
+
+from pulp_ansible.tests.functional.constants import (
+    ANSIBLE_COLLECTION_FILE_NAME,
+    ANSIBLE_COLLECTION_UPLOAD_FIXTURE_URL,
+    COLLECTION_METADATA,
+    GALAXY_ANSIBLE_BASE_URL_V3,
+)
+from pulp_ansible.tests.functional.utils import set_up_module as setUpModule  # noqa:F401
+
+from orionutils.generator import build_collection
+
+
+logger = logging.getLogger(__name__)
+
+
+repo = "automation-hub"
+
+
+def upload_handler(client, response):
+    """???
+
+    ???
+    """
+    response.raise_for_status()
+    logger.debug("response status: %s", response.status_code)
+    if response.status_code == 204:
+        return response
+    api._handle_202(client._cfg, response, client.pulp_host)
+    if response.request.method == "POST":
+        task_url = response.json()['task']
+        task = client.get(task_url)
+        return task
+    else:
+        return response.json()
+
+
+@pytest.fixture(scope='session')
+def artifact():
+    artifact = build_collection('skeleton')
+    return artifact
+
+
+@pytest.fixture(scope='session')
+def collection_upload(pulp_client, artifact):
+    # Publish a new collection
+    cfg = config.get_config()
+    UPLOAD_PATH = urljoin(cfg.get_base_url(), f"api/{repo}/v3/artifacts/collections/")
+    collection = {
+        "file": (ANSIBLE_COLLECTION_FILE_NAME, open(artifact.filename, 'rb'))
+    }
+
+    response = pulp_client.using_handler(upload_handler).post(UPLOAD_PATH, files=collection)
+    return response
+
+
+@pytest.fixture(scope='session')
+def collection_detail(collection_upload, pulp_client, artifact):
+    url = f"/api/{repo}/v3/collections/{artifact.namespace}/{artifact.name}/"
+    response = pulp_client.using_handler(api.json_handler).get(url)
+    return response
+
+
+@pytest.fixture(scope='session')
+def pulp_client():
+    cfg = config.get_config()
+    delete_orphans(cfg)
+    client = api.Client(cfg)
+    headers = cfg.custom['headers']
```

Comment:
What if `settings,json` has no `custom['headers']`?

Or we do a `cfg.custom.get("headers", "some default value")`

or

```python
try:
    headers = cfg.custom['headers']
except KeyError:
    pytest.exit("pulp-smash settings file is missing custom['headers'] attribute.")
```